docs/style-guide.md

```diff
@@ -0,0 +1,80 @@
+
+## Instructional Content
+
+The instructions for a step challenge should be as concise as possible to inform the camper what to do, with the exception of steps that introduce a new concept. When introducing a new concept, you'll generally use the following format:
+
+```md
+Definition
+
+Explanation
+
+Instructions
+```
+
+As an example:
+
+```md
+The <dfn>`alt`</dfn> attribute provides alternative text to display if the image fails to load, or to be read to assistive technology.
+
+The `alt` attribute should provide the same information in text form that a user would get if they visually saw the image.
+
+Add an `alt` attribute to your image with the value `An orange cat playing with a string`.
+```
+
+Breaking the text up like this prevents a wall of text, and increases the chance the camper will read.
+
+## Tests
+
+The tests for each step should be as granular as possible.
+
+Rather than a test that says `You should create a new div element with an id set to main and a class set to container`, you should have three separate tests:
+
+- You should create a new div element.
+- Your new div element should have an `id` attribute set to `main`.
+- Your new div element should have a `class` attribute set to `container`.
+
+This provides clarity for where a camper should look at their code to correct the issue.
+
+### How to Test Code
+
+If you are testing HTML, most likely you will want to query the `document` directly. The `document` object in the test runner refers specifically to the embedded preview of the camper's code, so you can query the DOM as you would in a browser.
+
+If you are testing CSS, you will typically want to use the CSS Helper. The CSS Helper is a module that allows you to query the CSS that the camper has written. It is available in the tests by calling `new __helpers.CSSHelp(document)`. <!-- TODO: separate docs for curriculum helper-->.
+
+If you are testing JavaScript, there are a few approaches you can take.
+
+First, global variables can be referenced directly in the tests. For example, if the camper writes the following:
+
+```js
+const myName = "Naomi";
+```
+
+You can directly assert the value:
+
+```js
+assert.strictEqual(myName, "Naomi");
+```
+
+Top-level functions can also be called directly in this way. If a function modifies the DOM, you can call the function and then query the DOM to assert the changes.
+
+For testing specific syntax (such as whether a variable was declared with `const`), you can run regular expressions against the `code` variable. This represents the user's written code.
+
+```js
+assert.match(code, /const\s+myName\s*)
+```
+
+A few caveats when running regular expressions against code:
+
+- Use capture groups to match quote pairs. `/const\s+myName\s*=\s*('|"|`)Naomi\1`
+- Calling `.toString()` on a function returns the Babel-ified code, which may not match the original code the camper wrote.
+- Be lenient on allowing valid syntax, such as spacing requirements.
```

Comment:
I would also mentioned that the tests should be lenient with semicolons since that has been an issue in the past